### PR TITLE
Update Java documentation

### PR DESCRIPTION
**Pull request by drodriguezhdez**

### CODE REVIEW

**Comment by drodriguezhdez** [comment]
Now. Check again, pleaase.

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,5 @@ website/build/
 website/yarn.lock
 website/node_modules
 website/i18n/*
+
+.idea/**
```
Comment (fermayo) [inline .gitignore]: Let's not commit this - the `.idea` folder can be ignored globally in your local git settings

File: docs/java-installation.md [hunk 2]
```diff
@@ -4,15 +4,17 @@ title: Scope Java Agent instructions
 sidebar_label: Installation
 ---
 
-Installation is done via [Maven](https://maven.apache.org/). Add the Scope agent dependency and version property to your `pom.xml` file,
-replacing `0.1.9` with the latest version of the agent:
+## Installation
```
Comment (fermayo) [inline docs/java-installation.md]:
```suggestion
```

We are already in the `Installation` section

File: docs/java-installation.md [hunk 3]
```diff
@@ -4,15 +4,17 @@ title: Scope Java Agent instructions
...
+### Maven
```
Comment (fermayo) [inline docs/java-installation.md]:
```suggestion
## Using Maven
```

File: docs/java-installation.md [hunk 4]
```diff
@@ -21,59 +23,59 @@ replacing `0.1.9` with the latest version of the agent:
 </dependency>
 ```
 
+### Gradle
```
Comment (fermayo) [inline docs/java-installation.md]:
```suggestion
## Using Gradle
```

File: docs/java-installation.md [hunk 5]
```diff
@@ -21,59 +23,59 @@ replacing `0.1.9` with the latest version of the agent:
...
 
-To use instrument your tests, configure the [`Maven Surefire Plugin`](https://maven.apache.org/surefire/maven-surefire-plugin/) 
-and/or the [`Maven Failsafe Plugin`](https://maven.apache.org/surefire/maven-failsafe-plugin/) to use the Scope Agent as a Java agent:
+dependencies {
+    testAgent "com.undefinedlabs.scope:scope-agent:0.1.9"
+}
+```
 
-```xml
+## Usage
```
Comment (fermayo) [inline docs/java-installation.md]:
```suggestion
## Instrumenting your tests
```

File: docs/java-installation.md [hunk 6]
```diff
@@ -21,59 +23,59 @@ replacing `0.1.9` with the latest version of the agent:
...
+### Maven
```
Comment (fermayo) [inline docs/java-installation.md]:
```suggestion

### Using Maven
```

File: docs/java-installation.md [hunk 7]
```diff
@@ -21,59 +23,59 @@ replacing `0.1.9` with the latest version of the agent:
...
+Configure the [`Maven Surefire Plugin`](https://maven.apache.org/surefire/maven-surefire-plugin/) and/or the [`Maven Failsafe Plugin`](https://maven.apache.org/surefire/maven-failsafe-plugin/) to use Scope agent as a Java agent:
+```
```
Comment (fermayo) [inline docs/java-installation.md]:
```suggestion
```xml
```

File: docs/java-installation.md [hunk 8]
```diff
@@ -21,59 +23,59 @@ replacing `0.1.9` with the latest version of the agent:
...
+           <goal>verify</goal>
+        </goals>
+      </execution>
+  </executions>
 </plugin>
 ```
-
 After this, you can run your tests as you normally do, for example using the `mvn clean verify` command.
 
+### Gradle
```
Comment (fermayo) [inline docs/java-installation.md]:
```suggestion
### Using Gradle
```

File: docs/java-installation.md [hunk 9]
```diff
@@ -21,59 +23,59 @@ replacing `0.1.9` with the latest version of the agent:
...
-
-Modify your Java app startup script to add the `javaagent` parameter pointing to the downloaded Scope Agent JAR.
-
-```bash
-java -javaagent:/path/to/scope/agent/scope-agent.jar -jar /path/to/app/my-app.jar
-``` 
-
-Notice that it is needed that `$SCOPE_APIKEY`, `$SCOPE_API_ENDPOINT`, `$SCOPE_REPOSITORY`, and `$SCOPE_COMMIT_SHA` 
-had been set as environment variables in the executing environment.
+```
```
Comment (fermayo) [inline docs/java-installation.md]:
```suggestion
```groovy
```
?

File: docs/java-runtime.md [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+---
+
+In order to see trace information from your Java service on integration and end-to-end tests,
+you need to use the Scope agent to instrument your running service.
+
+This service might run, for example, in a container on CI, or in a QA/staging environment.
+
+## Using `javaagent` VM param 
+
+Download the Scope Java Agent, changing version if it is necessary.
```
Comment (fermayo) [inline docs/java-runtime.md]:
```suggestion
Download the Scope Java Agent, replacing `0.1.9` with the latest version.
```

File: docs/java-runtime.md [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+```bash
+cd /path/to/scope/agent
+curl -s -o scope-agent.jar "https://repo1.maven.org/maven2/com/undefinedlabs/scope/scope-agent/0.1.9/scope-agent-0.1.9.jar"
+```
+
+Create / Modify `scope.yml` on the app root folder to disable testing mode
```
Comment (fermayo) [inline docs/java-runtime.md]: Testing mode should be disabled by default, right? And it should be automatically enabled if we detect we are running on CI?

File: .gitignore (hunk 1 above)
Comment (drodriguezhdez) [reply .gitignore]: Removed

File: docs/java-runtime.md (hunk 11 above)
Comment (drodriguezhdez) [reply docs/java-runtime.md]: Right. I'm removing this part.

File: docs/java-installation.md [hunk 12]
```diff
@@ -4,15 +4,16 @@ title: Scope Java Agent instructions
...
 ---
 
-Installation is done via [Maven](https://maven.apache.org/). Add the Scope agent dependency and version property to your `pom.xml` file,
-replacing `0.1.9` with the latest version of the agent:
+## Using Maven
 
-```xml
+Add the Scope agent dependency and version property to your `pom.xml` file, replacing `0.1.9` with the latest version of the agent:
+
+```
```
Comment (fermayo) [inline docs/java-installation.md]:
```suggestion
```xml
```

File: docs/java-installation.md [hunk 13]
```diff
@@ -4,15 +4,16 @@ title: Scope Java Agent instructions
...
 <properties>
   <scope.agent.version>0.1.9</scope.agent.version>
 </properties>
 ```
-```xml
+```
```
Comment (fermayo) [inline docs/java-installation.md]:
```suggestion
```xml
```

is there a reason why you removed the `xml` tag here?

File: docs/java-installation.md [hunk 14]
```diff
@@ -21,59 +22,59 @@ replacing `0.1.9` with the latest version of the agent:
 </dependency>
 ```
 
+## Using Gradle
+Add the `testAgent` entry to the `configurations` task block and add the Scope agent dependency, replacing 0.1.9 with the latest version of the agent.
 
-## Instrumenting your tests
+```
```
Comment (fermayo) [inline docs/java-installation.md]:
```suggestion
```groovy
```

**Pull request merged**